File: ruby/vertex.rb
require_relative 'hash_table'

class Vertex
  attr_reader :value, :edges
  
  def initialize(location)
    @edges = HashTable.new
    @value = location
  end

  def add_edge(edge)
    edges.insert(edge.id, edge)
  end

  def find_edge(edge_id)
    edges.find(edge_id)
  end

  # def find_closest_neighbor_with_available_packages(truck)
  #   edges.find { |e| e.available_packages?(truck) }
  # end

  # def closest_neighbor
  #   edges.first
  # end
end

File: python/vertex.py
from hash_table import HashTable

class Vertex(object):
    def __init__(self, location):
        self.edges = HashTable()
        self.value = location

    # add edge to hash table: O(n)
    def add_edge(self, edge):
        self.edges.insert(edge.identifier, edge)

    # find edge from id: O(n)
    def find_edge(self, edge_id):
        return self.edges.find(edge_id)

    # find distance to a neighboring vertex: O(n)
    def distance_to(self, location):
        return self.edges.find(location.identifier).weight
